h/search/query.py

```diff
@@ -46,12 +54,25 @@ def build(self, params):
         for key, value in params.items():
             matchers.append({"match": {key: value}})
 
-        query = {"match_all": {}}
+        bool_query = {}
 
+        # Add clauses executing in a query (scoring) context.
         if matchers:
-            query = {"bool": {"must": matchers}}
+            bool_query["must"] = matchers
```

Comment:
I understand what this is doing now but I found all these if else statements hard to follow. Could we simplify these/re-org them in a way that makes it more readable? Alternatively it might be better just to have a separate builder for es1 and es6?